File: Clients/ASystemModClient.pas
{**
@Abstract ASystem
@Author Prof1983 <prof1983@ya.ru>
@Created 19.11.2012
@LastMod 23.04.2013
}
unit ASystemModClient;

interface

uses
  ABase, ARuntime, ARuntimeBase, ASystemBase, ASystemProcSet;

function ASystem_Boot(): AError;

implementation

function ASystem_Boot(): AError;
var
  Module: AModule_Type;
begin
  if (ARuntime.Modules_GetByUid(ASystem_Uid, Module) < 0) then
  begin
    Result := -2;
    Exit;
  end;
  if Assigned(Module.GetProc) then
  begin
    Result := ASystem_SetProcs(Module.GetProc);
    Exit;
  end;

  {$ifdef ADepr}
  Result := ASystem_SetProcsP(Module.Procs);
  {$else}
  Result := -3;
  {$endif}
end;

end.
 

File: System/ASystemModClient.c
/*
Author Prof1983 <prof1983@ya.ru>
Created 15.04.2013
LastMod 15.04.2013
*/

#include "ABase.h"
#include "ARuntime.h"
#include "ASystemClient.c"

/*
  ARuntimeBase,
  ARuntimeMain,
  ASystemBase,
*/

const ASystem_Uid = 0x07082001;

AError
afunc ASystem_Boot() {
    AModule_Type Module;

    if (ARuntime_GetModuleByUid(ASystem_Uid, &Module) < 0) {
        return -2;
    }

    if (Module.GetProc == 0) {
        return -3;
    }
    return ASystem_LoadProcs(Module.GetProc);
}
